jung-algorithms/src/test/java/edu/uci/ics/jung/layout/spatial/FRLayoutsTest.java

```diff
@@ -109,9 +111,11 @@ public static void check() {
     log.debug("mapTwo:{}", mapTwo);
     log.debug("mapThree:{}", mapThree);
 
-    Assert.assertTrue(
-        "the compared maps are not close enough: mapTwo:" + mapTwo + ", mapThree:" + mapThree,
-        closeEnough(mapTwo, mapThree));
+    assertWithMessage(
+            "The compared maps are not close enough: mapTwo: %s, mapThree: %s", mapTwo, mapThree)
+        .that(mapTwo)
+        .comparingValuesUsing(Correspondence.tolerance(1.0E-4))
+        .equals(mapThree);
```

Comment:
I think this invocation of `.equals` won't do as you expect, because `Subject.equals` does the same thing as `Object.equals` (i.e. return true if `mapTwo` is equal to `mapThree`; false otherwise), whereas what I think you're after is functionality like `Assert.assertEquals`. Is my observation correct?

If so, you should replace this and all other calls to `Subject.equals` with `isEqualTo`. The JavaDoc for `Subject.equals` describes that:

> `Object.equals(Object)` is not supported on Truth subjects. If you meant to test object equality between an expected and the actual value, use `isEqualTo(Object)` instead.

Reply:
Good catch.
I was lulled into thinking it was easier than it is. For one, I am comparing Point instances, so I would probably need a Subject for Point. The compiler will not (and should not) let me just swap out isEqualTo and equals.

Reply:
Maybe error-prone would catch this at compile-time, if/once we start using it? I don't know if error-prone catches anything Truth-related, but the error-prone team have implemented checks specifically for Google-originated open source projects, so I wouldn't be surprised if a check for this appears eventually if it hasn't yet.